src/runs/execution/event-handlers/streaming.ts

```diff
@@ -317,6 +330,86 @@ export function createStreamingHandler(ctx: AgentContext) {
   };
 }
 
+export function createStreamlitStreamingHandler(ctx: AgentContext) {
+  return (emitter: Emitter<StreamlitEvents>) => {
+    emitter.on('newToken', async ({ delta }) => {
```

Comment:
Can we reuse this code from the createBeeStreamingHandler?

Reply:
I haven't found a good way to do it, the agent are different just enough that a bit of duplication makes sense imo.